resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/submit/ClientV2Suite.scala

```diff
@@ -169,30 +187,60 @@ class ClientV2Suite extends SparkFunSuite with BeforeAndAfter {
             .endMetadata()
         }
       })
-    when(initContainerComponentsProvider.provideContainerLocalizedFilesResolver())
-      .thenReturn(containerLocalizedFilesResolver)
+    when(initContainerComponentsProvider.provideContainerLocalizedFilesResolver(
+      any[String])).thenReturn(containerLocalizedFilesResolver)
+    when(initContainerComponentsProvider.provideDriverPodFileMounter())
+        .thenReturn(fileMounter)
     when(submittedDependenciesSecretBuilder.build())
       .thenReturn(INIT_CONTAINER_SECRET)
     when(kubernetesClient.pods()).thenReturn(podOps)
     when(podOps.create(any())).thenAnswer(new Answer[Pod] {
       override def answer(invocation: InvocationOnMock): Pod = {
         new PodBuilder(invocation.getArgumentAt(0, classOf[Pod]))
           .editMetadata()
-          .withUid(DRIVER_POD_UID)
+            .withUid(DRIVER_POD_UID)
           .endMetadata()
-          .withKind(DRIVER_POD_KIND)
+            .withKind(DRIVER_POD_KIND)
           .withApiVersion(DRIVER_POD_API_VERSION)
           .build()
       }
     })
     when(podOps.withName(s"$APP_RESOURCE_PREFIX-driver")).thenReturn(namedPodResource)
+    when(fileMounter.addPySparkFiles(
+      mockitoEq(RESOLVED_PYSPARK_PRIMARY_FILE),
+      mockitoEq(RESOLVED_PYSPARK_FILES.mkString(",")),
+      any[String],
+      any())).thenAnswer( new Answer[PodBuilder] {
+        override def answer(invocation: InvocationOnMock) : PodBuilder = {
+          invocation.getArgumentAt(3, classOf[PodBuilder])
+          .editSpec()
+            .editMatchingContainer(new ContainerNameEqualityPredicate(
+              invocation.getArgumentAt(2, classOf[String])))
+              .addNewEnv()
```

Comment:
This is still too complex - just add to the labels or the annotations in the metadata.